### PR TITLE
HWKAPM-630 Distributed trace 'root node' dropdown should list all ini…

### PR DESCRIPTION
**Pull request by objectiser**
…tial endpoints invoked to create trace instances

### CODE REVIEW

**Comment by objectiser** [comment]
@pavolloffay Could you review please?

Reviewer notes:
* Instead of calculating the set of root nodes based on which graph nodes had no 'clients', the root nodes are now determined based on the top level endpoints invoked - obtained from the trace completion times.
* Changed the dropdown label in the UI to be 'endpoints' as it seems more appropriate.

Can you check against the polyglot example.

File: ui/src/main/scripts/plugins/e2e/html/e2e.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <div class="card-pf-body">
       <form class="form-horizontal form-inline row">
         <div class="form-group col-md-5">
-          <label class="control-label col-md-3" for="rootnode">Root Node </label>
+          <label class="control-label col-md-3" for="rootnode">Endpoint </label>
```
Comment (pavolloffay) [inline ui/src/main/scripts/plugins/e2e/html/e2e.html]: Could we rename it to `Root Endpoint` - as it represents only root endpoints.

Comment (objectiser) [inline ui/src/main/scripts/plugins/e2e/html/e2e.html]: Would prefer "Initial Endpoint"?  - root doesn't really have any meaning to the end user.

Comment (pavolloffay) [inline ui/src/main/scripts/plugins/e2e/html/e2e.html]: Yes that sounds good

**Comment by pavolloffay** [comment]
Run it again and it seems ok, however: 

1. start clean APM server,
2. start polyglot-zipkin

it will show 0 Root nodes and 10 instance details, when you click on details it throws exception/

**Comment by objectiser** [comment]
Yes found that aswell - it is the initial fragments that get reported when the docker-compose up is performed, before any real interactions are performed. This is a separate issue that needs investigation: https://issues.jboss.org/browse/HWKAPM-637

**Comment by pavolloffay** [comment]
But it should not show any instances without root node

**Comment by objectiser** [comment]
@pavolloffay Thanks! The other issues will be handled in that other jira.

**Pull request merged**